service/controller/app/v1/resource/chart/desired.go

```diff
@@ -30,7 +27,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	config := generateConfig(cr, cc.AppCatalog, r.chartNamespace)
-	tarballURL, err := generateTarballURL(appcatalogkey.AppCatalogStorageURL(cc.AppCatalog), key.AppName(cr), key.Version(cr))
+	tarballURL, err := appcatalogkey.GenerateTarballURL(appcatalogkey.AppCatalogStorageURL(cc.AppCatalog), key.AppName(cr), key.Version(cr))
```

Comment:
Why is this is in `appcatalogkey` and not `key`? 

It's only needed for the `app` controller isn't it? If so it would be good to move it.